File: clients/Java/src/main/java/model/Vec2Int.java
package model;

import util.StreamUtil;

public class Vec2Int {
    private int x;
    public int getX() { return x; }
    public void setX(int x) { this.x = x; }
    private int y;
    public int getY() { return y; }
    public void setY(int y) { this.y = y; }
    public Vec2Int() {}
    public Vec2Int(int x, int y) {
        this.x = x;
        this.y = y;
    }
    public static Vec2Int readFrom(java.io.InputStream stream) throws java.io.IOException {
        Vec2Int result = new Vec2Int();
        result.x = StreamUtil.readInt(stream);
        result.y = StreamUtil.readInt(stream);
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, x);
        StreamUtil.writeInt(stream, y);
    }
}

File: clients/Scala/src/main/scala/model/Vec2Int.scala
package model

import util.StreamUtil

case class Vec2Int(x: Int, y: Int) {
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, x)
        StreamUtil.writeInt(stream, y)
    }
}
object Vec2Int {
    def readFrom(stream: java.io.InputStream): Vec2Int = Vec2Int(
        StreamUtil.readInt(stream)
        ,
        StreamUtil.readInt(stream)
        )
}
